client.go

```diff
@@ -476,6 +476,24 @@ func (a *Agent) JSON(v interface{}) *Agent {
 	return a
 }
 
+// CustomJSON sends a JSON request with the content header
+// set to the given ctype string.
+func (a *Agent) CustomJSON(v interface{}, ctype string) *Agent {
```

Comment:
why not do something like this ?

```go
func (a *Agent) CustomJSON(v interface{}, ctype string) *Agent {
     a.JSON(v)
     a.req.Header.SetContentType(ctype)
     return a
}
```

Reply:
That works. I wasn't sure if we wanted to set the Content-Header twice, once within `JSON` and once within `CustomJSON`, but this would definitely simplify things. Changed.